Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Fri, 13 Feb 2026 23:13:04 -0300
Last-Update: 2026-02-13
Subject: [PATCH] add buffer overflow patch
---
 .SRCINFO |    2 +-
 PKGBUILD |    5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = dash
 	pkgdesc = POSIX compliant shell that aims to be as small as possible
 	pkgver = 0.5.13.2
-	pkgrel = 0.1
+	pkgrel = 0.2
 	url = http://gondor.apana.org.au/~herbert/dash/
 	install = dash.install
 	arch = x86_64
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -3,7 +3,7 @@
 
 pkgname=dash
 pkgver=0.5.13.2
-pkgrel=0.1
+pkgrel=0.2
 _commit=5139d6a27763fc0be386e15634db21e45598b299
 pkgdesc='POSIX compliant shell that aims to be as small as possible'
 url='http://gondor.apana.org.au/~herbert/dash/'
@@ -32,7 +32,8 @@ _cflags() {
 
 build() {
   cd ${pkgname}-${_commit}
-  export CFLAGS="$(_cflags x86-64 generic)"
+  export CFLAGS="$(_cflags x86-64 generic) -g -fsanitize=undefined -fsanitize=address"
+  export LDFLAGS="${LDFLAGS} -fsanitize=undefined -fsanitize=address"
   ./configure \
     --prefix=/usr \
     --bindir=/usr/bin \
