Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Tue, 4 Nov 2025 07:59:00 -0300
Last-Update: 2026-07-28
Subject: [PATCH] See <https://github.com/thelounge/thelounge/pull/5003>
---
 .SRCINFO   |    4 +++-
 .gitignore |    1 +
 PKGBUILD   |    5 ++++-
 3 files changed, 8 insertions(+), 2 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = thelounge-beta
 	pkgdesc = Modern self-hosted web IRC client (Latest release/pre-release)
 	pkgver = 4.6.0pre.1
-	pkgrel = 1.1
+	pkgrel = 1.2
 	url = https://thelounge.chat/
 	arch = any
 	license = MIT
@@ -14,11 +14,13 @@ pkgbase = thelounge-beta
 	options = !lto
 	backup = etc/thelounge/config.js
 	source = thelounge-438525ff10281b07864b3c7456d151d7e96b0032.tar.gz::https://github.com/thelounge/thelounge/archive/438525ff10281b07864b3c7456d151d7e96b0032.tar.gz
+	source = https://github.com/thelounge/thelounge/pull/5003.patch
 	source = system.service
 	source = user.service
 	source = sysusers.d
 	source = tmpfiles.d
 	sha256sums = 9c7db14ad7a8e2a8207ab07a187813045d7267efdc89eaa654f1dc7d573615d5
+	sha256sums = 311307583dc9098bb1e08d6fb763232f791fec34b8b331230f09857ff35fd013
 	sha256sums = c92210f6ac8f01c1cd01b6b26793094cd2feea583ed21fab3564d6bcafdc7a20
 	sha256sums = c609f3309f54bd6285e99ff29ca2464828bec7bbbca67243ee688bd2d605dbf0
 	sha256sums = 30fab63b8a4ffcfdda4c5b8d7c66822a323c4f1de6ca62b77fe9500f4befc0a5
Index: b/.gitignore
===================================================================
--- a/.gitignore
+++ b/.gitignore
@@ -2,3 +2,4 @@
 *.tar.gz
 *.tgz
 *.log
+5003.patch
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@ _pkgname=thelounge
 pkgname=thelounge-beta
 _pkgver=4.6.0-pre.1
 pkgver=${_pkgver/-/}
-pkgrel=1.1
+pkgrel=1.2
 pkgdesc='Modern self-hosted web IRC client (Latest release/pre-release)'
 url='https://thelounge.chat/'
 arch=('any')
@@ -19,12 +19,14 @@ backup=('etc/thelounge/config.js')
 _commit='438525ff10281b07864b3c7456d151d7e96b0032'
 source=(
     "thelounge-${_commit}.tar.gz::https://github.com/thelounge/thelounge/archive/${_commit}.tar.gz"
+    'https://github.com/thelounge/thelounge/pull/5003.patch'
     'system.service'
     'user.service'
     'sysusers.d'
     'tmpfiles.d'
 )
 sha256sums=('9c7db14ad7a8e2a8207ab07a187813045d7267efdc89eaa654f1dc7d573615d5'
+            '311307583dc9098bb1e08d6fb763232f791fec34b8b331230f09857ff35fd013'
             'c92210f6ac8f01c1cd01b6b26793094cd2feea583ed21fab3564d6bcafdc7a20'
             'c609f3309f54bd6285e99ff29ca2464828bec7bbbca67243ee688bd2d605dbf0'
             '30fab63b8a4ffcfdda4c5b8d7c66822a323c4f1de6ca62b77fe9500f4befc0a5'
@@ -33,6 +35,7 @@ sha256sums=('9c7db14ad7a8e2a8207ab07a187
 prepare() {
     cd "${_pkgname}-${_commit}"
     echo /etc/thelounge > .thelounge_home
+    patch -t -Np1 -i ../5003.patch
     yarn install --frozen-lockfile --non-interactive --ignore-scripts --cache-folder "$srcdir/yarn-cache"
 }
 
