Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Thu, 5 Feb 2026 23:59:22 -0300
Last-Update: 2026-07-27
Subject: [WIP] Split PSW package groups
---
 PKGBUILD |  194 ++++++++++++++++++++++++++++++++++++++++++++++++---------------
 1 file changed, 149 insertions(+), 45 deletions(-)

Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,7 +1,14 @@
 # Maintainer: marmis <tiagodepalves@gmail.com>
 # Contributor: "marmis" Tiago de Paula <tiagodepalves@gmail.com>
 
-pkgname=intel-sgx-psw-bin
+pkgbase=intel-sgx-psw-bin
+pkgname=(
+  intel-sgx-psw-common-bin
+  intel-sgx-aesm-bin
+  intel-sgx-qgs-bin
+  intel-sgx-dcap-bin
+  intel-sgx-ra-bin
+)
 pkgdesc='Intel® Software Guard Extensions Platform Software for Linux* OS'
 pkgver=2.29
 pkgrel=1
@@ -9,13 +16,6 @@ url='https://github.com/intel/confidenti
 arch=('x86_64')
 license=('BSD-3-Clause AND LicenseRef-Intel-SGX-Third-Party') # https://github.com/intel/confidential-computing.sgx/blob/main/License.txt
 makedepends=()
-depends=('glibc' 'gcc-libs' 'bash')
-optdepends=(
-  'protobuf-21: required for the AESM service'
-  'nodejs: for running the SGX DCAP PCCS Server'
-  'curl: required for SGX Remote Attestation Service (RA)'
-  'boost190: required by the TDX Quote Generation Service (QGS)'
-)
 provides=("intel-sgx-psw=${pkgver}")
 conflicts=('intel-sgx-psw')
 options=(!strip !debug) # debug symbols already included from upstream, better not to strip them
@@ -25,55 +25,159 @@ source=("sgx_${pkgver}_debian_local_repo
 b2sums=('5832a7537c7bb0b50376bf6818e647bd738ba54da6263249ddf7c6a6483b60a0458bc7177c893ea7a364ddfa9c9ab9abcb5925ef12dd30c3f4ab61b9c0d62180'
         'f0a1415f8e88ffe2fe189c56a701a9756f74947375f142aa3965679129d3182f1b380beea02d70e07cbe7c9443461b8cbb674193ae07f71d543d13a10ecaf7ff')
 
-# Move files to standard Arch Linux folders
-_fix_debian_paths() {
+# Move files to standard Arch Linux folders (namcap rule: directoryname)
+_fix_non_standard_paths() {
   # /etc/udev is for user files, packages should use /usr/lib/udev instead
-  install -d -m755 "${pkgdir}"/usr/lib/udev
-  mv "${pkgdir}"/etc/udev/rules.d "${pkgdir}"/usr/lib/udev/rules.d
-  rmdir "${pkgdir}"/etc/udev
-
-  # Debian has a /usr/lib/$TARGET folder, but you won't find them in Arch
-  mv "${pkgdir}"/usr/lib/"${CARCH}"-linux-gnu/*.so* "${pkgdir}"/usr/lib/
-  rmdir "${pkgdir}"/usr/lib/"${CARCH}"-linux-gnu
+  if [[ -d "${pkgdir}"/etc/udev ]]; then
+    install -d -m755 "${pkgdir}"/usr/lib/udev
+    mv "${pkgdir}"/etc/udev/rules.d "${pkgdir}"/usr/lib/udev/rules.d
+    rmdir "${pkgdir}"/etc/udev
+  fi
+
+  # Debian has a /usr/lib/$TARGET folder, but not in Arch
+  if [[ -d "${pkgdir}"/usr/lib/"${CARCH}"-linux-gnu ]]; then
+    mv "${pkgdir}"/usr/lib/"${CARCH}"-linux-gnu/*.so* "${pkgdir}"/usr/lib/
+    rmdir "${pkgdir}"/usr/lib/"${CARCH}"-linux-gnu
+  fi
 
   # Debian has an actual /lib folder, which in Arch is just a symlink to /usr/lib
-  mv "${pkgdir}"/lib/systemd/system/*.service "${pkgdir}"/usr/lib/systemd/system/
-  rmdir "${pkgdir}"/lib/systemd/system "${pkgdir}"/lib/systemd "${pkgdir}"/lib
+  if [[ -d "${pkgdir}"/lib/systemd/system ]]; then
+    mv "${pkgdir}"/lib/systemd/system/*.service "${pkgdir}"/usr/lib/systemd/system/
+    rmdir "${pkgdir}"/lib/systemd/system "${pkgdir}"/lib/systemd "${pkgdir}"/lib
+  fi
+
+  # Debian also supports /include, which should be empty here
+  if [[ -d "${pkgdir}"/include ]]; then
+    rmdir "${pkgdir}"/include
+  fi
+}
+
+# Update symbolic links after moving to standard Arch Linux paths (namcap rule: symlink)
+_fix_symlink() {
+  local target
+  target="${pkgdir}/usr/lib/$(basename "${1}")"
+  target="$(realpath --relative-to="$(dirname "${1}")" "${target}")"
+  ln -sf "${target}" "${1}"
+}
+
+_pick() {
+  find "${srcdir}"/sgx_debian_local_repo \( -name "${1}.deb" -or -name "${1}-dbgsym.deb" \) \
+    -exec sh -c 'ar -p "$1" -O data.tar.zst | tar -x --zstd -C "$0"' "${pkgdir}" {} \;
+}
+
+package_intel-sgx-psw-common-bin() {
+  pkgdesc='Intel® SGX Platform Software (PSW) for Linux* OS'
+  depends=('glibc' 'gcc-libs' 'protobuf-21')
+  conflicts=('intel-sgx-psw-common')
+
+  _pick libsgx-enclave-common
+  _pick libsgx-enclave-common-dev
+  _pick libsgx-headers
+  _pick libsgx-launch
+  _pick libsgx-launch-dev
+  _pick libsgx-urts
+  _fix_non_standard_paths
 
-  # Debian also supports /include, which is empty here
-  rmdir "${pkgdir}"/include
+  install -Dm644 intel-sgx-sysusers.conf "${pkgdir}"/usr/lib/sysusers.d/intel-sgx.conf
+  install -Dm644 "${pkgdir}"/usr/share/doc/libsgx-enclave-common/copyright -t "${pkgdir}/usr/share/licenses/intel-sgx-psw-bin"
+}
+
+package_intel-sgx-aesm-bin() {
+  pkgdesc='Intel® SGX Application Enclave Services Manager (AESM)'
+  depends=('intel-sgx-psw-common-bin' 'glibc' 'gcc-libs' 'protobuf-21')
+  optdepends=('intel-sgx-qgs-bin: for the Untrusted AE (UAE) Service')
+  conflicts=('intel-sgx-aesm')
+
+  _pick sgx-aesm-service
+  _pick libsgx-uae-service
+  _pick libsgx-ae-epid
+  _pick libsgx-ae-id-enclave
+  _pick libsgx-ae-le
+  _pick libsgx-ae-pce
+  _pick libsgx-ae-qe3
+  _pick libsgx-ae-qve
+  _pick libsgx-aesm-ecdsa-plugin
+  _pick libsgx-aesm-epid-plugin
+  _pick libsgx-aesm-launch-plugin
+  _pick libsgx-aesm-pce-plugin
+  _pick libsgx-aesm-quote-ex-plugin
+  _pick libsgx-qe3-logic
+  _pick libsgx-epid-dev
+  _pick libsgx-epid
+  _pick libsgx-pce-logic
+  _fix_non_standard_paths
+
+  _fix_symlink "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_pce.signed.so"
+  _fix_symlink "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_pce.signed.so.1"
+  _fix_symlink "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_qe3.signed.so"
+  _fix_symlink "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_qe3.signed.so.1"
+
+  # TODO: install -Dm644 intel-sgx-aesm-sysusers.conf "${pkgdir}"/usr/lib/sysusers.d/intel-sgx-aesm.conf
 }
 
-# Update symbolic links using non-standard Arch Linux paths
-_fix_binary_symlinks() {
-  local broken_symlinks=(
-    "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_pce.signed.so"
-    "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_pce.signed.so.1"
-    "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_qe3.signed.so"
-    "${pkgdir}/opt/intel/sgx-aesm-service/aesm/libsgx_qe3.signed.so.1"
-    "${pkgdir}/opt/intel/sgx-pck-id-retrieval-tool/libsgx_id_enclave.signed.so.1"
-    "${pkgdir}/opt/intel/sgx-pck-id-retrieval-tool/libsgx_pce.signed.so.1"
+package_intel-sgx-qgs-bin() {
+  pkgdesc='Intel® SGX Quote Generation Service (QGS)'
+  depends=('intel-sgx-psw-common-bin' 'glibc' 'gcc-libs' 'protobuf-21' 'boost183')
+  optdepends=(
+    'intel-sgx-aesm-bin: for Trust Domain Extension (TDX) QE logic'
+    'intel-sgx-dcap-bin: for Trust Domain Extension (TDX) QE logic'
   )
+  conflicts=('intel-sgx-qgs')
 
-  local symlink target
-  for symlink in "${broken_symlinks[@]}"; do
-    target="${pkgdir}/usr/lib/$(basename "${symlink}")"
-    target="$(realpath --relative-to="$(dirname "${symlink}")" "${target}")"
+  _pick tdx-qgs
+  _pick libsgx-tdx-logic
+  _pick libsgx-tdx-logic-dev
+  _pick libsgx-quote-ex
+  _pick libsgx-quote-ex-dev
+  _pick libsgx-ae-tdqe
+  _pick libtdx-attest
+  _pick libtdx-attest-dev
+  _fix_non_standard_paths
 
-    ln -sf "${target}" "${symlink}"
-  done
+  # TODO: install -Dm644 intel-sgx-qgs-sysusers.conf "${pkgdir}"/usr/lib/sysusers.d/intel-sgx-qgs.conf
 }
 
-package() {
-  find "${srcdir}"/sgx_debian_local_repo -name "*.deb" -exec \
-    sh -c 'ar -p "$1" -O data.tar.zst | tar -x --zstd -C "$0"' "${pkgdir}" {} \;
+package_intel-sgx-dcap-bin() {
+  pkgdesc='Intel® SGX Data Center Attestation Primitives (DCAP)'
+  depends=('intel-sgx-psw-common-bin' 'intel-sgx-awsm-bin' 'glibc' 'gcc-libs' 'curl')
+  optdepends=( # SPLIT?
+    'bash: for the Provisioning Certificate Caching Service'
+    'python: for the Provisioning Certificate Caching Service'
+    'python-pip: Provisioning Certificate Caching Service'
+    'python-venv: Provisioning Certificate Caching Service'
+    'npm: for the Provisioning Certificate Caching Service'
+    'nodejs: Provisioning Certificate Caching Service'
+  )
+  conflicts=('intel-sgx-dcap')
 
-  # required users and groups
-  install -Dm644 intel-sgx-sysusers.conf "${pkgdir}"/usr/lib/sysusers.d/intel-sgx.conf
+  _pick sgx-dcap-pccs
+  _pick libsgx-dcap-ql
+  _pick libsgx-dcap-ql-dev
+  _pick libsgx-dcap-default-qpl
+  _pick libsgx-dcap-default-qpl-dev
+  _pick libsgx-dcap-quote-verify
+  _pick libsgx-dcap-quote-verify-dev
+  _pick intel-tee-pccs-admin-tool
+  _pick intel-tee-pcs-client-tool
+  _pick sgx-pck-id-retrieval-tool
+  _pick tee-appraisal-tool
+  _fix_non_standard_paths
 
-  # composed license
-  install -Dm644 "${pkgdir}"/usr/share/doc/libsgx-enclave-common/copyright -t "${pkgdir}/usr/share/licenses/${pkgname}"
+  _fix_symlink "${pkgdir}/opt/intel/sgx-pck-id-retrieval-tool/libsgx_id_enclave.signed.so.1"
+  _fix_symlink "${pkgdir}/opt/intel/sgx-pck-id-retrieval-tool/libsgx_pce.signed.so.1"
+
+  # TODO: install -Dm644 intel-sgx-pccs-sysusers.conf "${pkgdir}"/usr/lib/sysusers.d/intel-sgx-pccs.conf
+}
 
-  _fix_debian_paths # namcap rule: directoryname
-  _fix_binary_symlinks # namcap rule: symlink
+package_intel-sgx-ra-bin() {
+  pkgdesc='Intel® SGX Remote Attestation (RA) Service'
+  depends=('glibc' 'gcc-libs')
+  conflicts=('intel-sgx-ra')
+
+  _pick sgx-ra-service
+  _pick libsgx-ra-network
+  _pick libsgx-ra-network-dev
+  _pick libsgx-ra-uefi
+  _pick libsgx-ra-uefi-dev
+  _fix_non_standard_paths
 }
