Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Mon, 5 Jan 2026 19:01:00 -0300
Last-Update: 2026-07-27
Subject: [PATCH] Patch for `derivePassword`.

From <https://github.com/go-sprout/sprout/pull/157>
---
 .SRCINFO |    6 ++++--
 PKGBUILD |   17 ++++++++++-------
 2 files changed, 14 insertions(+), 9 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = chezmoi
 	pkgdesc = Manage your dotfiles across multiple machines
 	pkgver = 2.71.1
-	pkgrel = 1.2
+	pkgrel = 1.3
 	url = https://www.chezmoi.io/
 	arch = x86_64
 	license = MIT
@@ -13,7 +13,9 @@ pkgbase = chezmoi
 	options = !lto
 	source = chezmoi-2.71.1.tar.gz::https://github.com/twpayne/chezmoi/archive/v2.71.1.tar.gz
 	source = https://github.com/twpayne/chezmoi/commit/a08a575a051dbe93b9dadb63f683e437af1337ba.patch
-	sha512sums = 744dba263b89d1dedd6c6f1bca002be7e40a56a20c6cfdfbc0fe0cf27b37393040681becb217432ef1c919639f7341a325a66dae87641e05b951e9c9defcbe45
+	source = https://github.com/go-sprout/sprout/pull/157.patch
+	sha512sums = 09eeeb1a55ec963751fdd4d1ccc0f4b0d7125b668665ba95bafdeda433e173882a439447052babacb73178e2afeb8af6eaf6ea7c343d7305e13f4c5c322ec7f3
 	sha512sums = 55176706b1429e675e668f1fbe6df15d810e3d0f4562b10eaf262b18f3ba28c614e8b73139c84870e3d4cc340d93330ef856642a2288afa2d4bc7b99f799b719
+	sha512sums = 3376d8dc6f7053c3d12bdb43d6b7734471626ee3255e99a0e91714227d1febd58739e2e4c93a3ca9fa932a559b28a728f13c0b3fbe8e9a71d016e95952e60479
 
 pkgname = chezmoi
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@
 
 pkgname=chezmoi
 pkgver=2.71.1
-pkgrel=1.2
+pkgrel=1.3
 pkgdesc="Manage your dotfiles across multiple machines"
 arch=('x86_64')
 url='https://www.chezmoi.io/'
@@ -16,9 +16,11 @@ checkdepends=('zip')
 optdepends=('git: clone remote dotfiles over ssh')
 options=('!lto')
 source=("${pkgname}-${pkgver}.tar.gz::https://github.com/twpayne/chezmoi/archive/v${pkgver}.tar.gz"
-        'https://github.com/twpayne/chezmoi/commit/a08a575a051dbe93b9dadb63f683e437af1337ba.patch')
-sha512sums=('744dba263b89d1dedd6c6f1bca002be7e40a56a20c6cfdfbc0fe0cf27b37393040681becb217432ef1c919639f7341a325a66dae87641e05b951e9c9defcbe45'
-            '55176706b1429e675e668f1fbe6df15d810e3d0f4562b10eaf262b18f3ba28c614e8b73139c84870e3d4cc340d93330ef856642a2288afa2d4bc7b99f799b719')
+        'https://github.com/twpayne/chezmoi/commit/a08a575a051dbe93b9dadb63f683e437af1337ba.patch'
+        'https://github.com/go-sprout/sprout/pull/157.patch')
+sha512sums=('09eeeb1a55ec963751fdd4d1ccc0f4b0d7125b668665ba95bafdeda433e173882a439447052babacb73178e2afeb8af6eaf6ea7c343d7305e13f4c5c322ec7f3'
+            '55176706b1429e675e668f1fbe6df15d810e3d0f4562b10eaf262b18f3ba28c614e8b73139c84870e3d4cc340d93330ef856642a2288afa2d4bc7b99f799b719'
+            '3376d8dc6f7053c3d12bdb43d6b7734471626ee3255e99a0e91714227d1febd58739e2e4c93a3ca9fa932a559b28a728f13c0b3fbe8e9a71d016e95952e60479')
 
 prepare() {
   cd "$pkgname-$pkgver"
@@ -28,7 +30,8 @@ prepare() {
   go mod edit -replace github.com/go-sprout/sprout=github.com/go-sprout/sprout@3b77ce5490dd49ac5164399823c8637ba4193f24
   go mod tidy
 
-  go mod download
+  go mod vendor
+  git -C vendor/github.com/go-sprout/sprout apply --reject "$srcdir/157.patch" || :
 }
 
 build() {
@@ -41,11 +44,11 @@ build() {
   export GOPROXY=off
   go build \
     -buildmode=pie \
-    -mod=readonly \
+    -mod=vendor \
     -tags noupgrade \
     -ldflags "-compressdwarf=false \
               -X github.com/twpayne/chezmoi/cmd.DocsDir=/usr/share/doc/$pkgname \
-              -X main.version=$pkgver+sprout-3b77ce54 \
+              -X main.version=$pkgver+sprout-3b77ce54.sprout-170 \
               -X main.date=$(date -u +'%Y-%m-%dT%H:%M:%SZ' --date=@$SOURCE_DATE_EPOCH) \
               -extldflags \"$LDFLAGS\"" \
     .
