Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Mon, 5 Jan 2026 19:01:00 -0300
Last-Update: 2026-08-19
Subject: [PATCH] Patch for `derivePassword`.

From <https://github.com/go-sprout/sprout/pull/157>
---
 .SRCINFO |    3 ++-
 PKGBUILD |   13 ++++++++-----
 2 files changed, 10 insertions(+), 6 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = chezmoi
 	pkgdesc = Manage your dotfiles across multiple machines
 	pkgver = 2.72.0
-	pkgrel = 1.1
+	pkgrel = 1.2
 	url = https://www.chezmoi.io/
 	arch = x86_64
 	license = MIT
@@ -12,6 +12,7 @@ pkgbase = chezmoi
 	optdepends = git: clone remote dotfiles over ssh
 	options = !lto
 	source = chezmoi-d969d39c82998301ca87582e27f01127cc626a65.tar.gz::https://github.com/twpayne/chezmoi/archive/d969d39c82998301ca87582e27f01127cc626a65.tar.gz
+	source = https://github.com/go-sprout/sprout/pull/157.patch
 	sha512sums = 080b7543497a5a1ed76b27d475d0be552785383c4887b50c79665644173efd24892ca03b3a0c88759e9c6349d801d44f5f072255f9e29f48300cd8e77b3cd7c8
 
 pkgname = chezmoi
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -16,13 +16,16 @@ checkdepends=('zip')
 optdepends=('git: clone remote dotfiles over ssh')
 options=('!lto')
 _commit=d969d39c82998301ca87582e27f01127cc626a65
-source=("${pkgname}-${_commit}.tar.gz::https://github.com/twpayne/chezmoi/archive/${_commit}.tar.gz")
+source=("${pkgname}-${_commit}.tar.gz::https://github.com/twpayne/chezmoi/archive/${_commit}.tar.gz"
+        'https://github.com/go-sprout/sprout/pull/157.patch')
 sha512sums=('080b7543497a5a1ed76b27d475d0be552785383c4887b50c79665644173efd24892ca03b3a0c88759e9c6349d801d44f5f072255f9e29f48300cd8e77b3cd7c8')
 
+
 prepare() {
   cd "$pkgname-$_commit"
 
-  go mod download
+  go mod vendor
+  git -C vendor/github.com/go-sprout/sprout apply --reject ../157.patch || :
 }
 
 build() {
@@ -35,11 +38,11 @@ build() {
   export GOPROXY=off
   go build \
     -buildmode=pie \
-    -mod=readonly \
+    -mod=vendor \
     -tags noupgrade \
     -ldflags "-compressdwarf=false \
               -X github.com/twpayne/chezmoi/cmd.DocsDir=/usr/share/doc/$pkgname \
-              -X main.version=$pkgver+sprout \
+              -X main.version=$pkgver+sprout.157 \
               -X main.date=$(date -u +'%Y-%m-%dT%H:%M:%SZ' --date=@$SOURCE_DATE_EPOCH) \
               -extldflags \"$LDFLAGS\"" \
     .
@@ -63,4 +66,4 @@ package() {
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
 
-pkgrel+=.1
+pkgrel+=.2
