---
 .SRCINFO   |    2 +-
 .gitignore |    2 ++
 PKGBUILD   |    5 ++++-
 3 files changed, 7 insertions(+), 2 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = chezmoi
 	pkgdesc = Manage your dotfiles across multiple machines
 	pkgver = 2.69.0
-	pkgrel = 1
+	pkgrel = 1.1
 	url = https://www.chezmoi.io/
 	arch = x86_64
 	license = MIT
Index: b/.gitignore
===================================================================
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,2 @@
+*.pkg.tar.*
+*.tar.gz
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -5,7 +5,7 @@
 
 pkgname=chezmoi
 pkgver=2.69.0
-pkgrel=1
+pkgrel=1.1
 pkgdesc="Manage your dotfiles across multiple machines"
 arch=('x86_64')
 url='https://www.chezmoi.io/'
@@ -21,6 +21,9 @@ sha512sums=('8fc2d5232914a9004377d14b239
 prepare() {
   cd "$pkgname-$pkgver"
 
+  go mod edit -replace github.com/go-sprout/sprout=github.com/twpayne/sprout@541b6bfa22450a3a9561506bbe8bc57dfba030df
+  go mod tidy
+
   go mod download
 }
 
