Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Wed, 03 Jun 2026 23:28:40 -0300
Last-Update: 2026-08-19
Subject: [PATCH] Update chezmoi to the latest version
---
 .SRCINFO   |    2 +-
 .gitignore |   11 +++++++++++
 PKGBUILD   |   16 +++++++++++-----
 REUSE.toml |   30 +++++++++++++++---------------
 4 files changed, 38 insertions(+), 21 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = chezmoi
 	pkgdesc = Manage your dotfiles across multiple machines
 	pkgver = 2.72.0
-	pkgrel = 1
+	pkgrel = 1.0
 	url = https://www.chezmoi.io/
 	arch = x86_64
 	license = MIT
Index: b/.gitignore
===================================================================
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,11 @@
+*
+# dev tools
+!.gitignore
+!.nvchecker.toml
+!LICENSE
+!LICENSES
+!LICENSES/*
+!REUSE.toml
+# actual package
+!PKGBUILD
+!.SRCINFO
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -32,15 +32,19 @@ build() {
   export CGO_CPPFLAGS="$CPPFLAGS"
   export CGO_CXXFLAGS="$CXXFLAGS"
   export GOPROXY=off
+  local go_ldflags=(
+    -compressdwarf=false
+    -X "'github.com/twpayne/chezmoi/cmd.DocsDir=/usr/share/doc/${pkgname}'"
+    -X "'main.version=${pkgver}'"
+    -X "'main.date=$(date -u +'%Y-%m-%dT%H:%M:%SZ' --date="@${SOURCE_DATE_EPOCH}")'"
+    -extldflags "'${LDFLAGS}'"
+  )
+
   go build \
     -buildmode=pie \
     -mod=readonly \
     -tags noupgrade \
-    -ldflags "-compressdwarf=false \
-              -X github.com/twpayne/chezmoi/cmd.DocsDir=/usr/share/doc/$pkgname \
-              -X main.version=$pkgver \
-              -X main.date=$(date -u +'%Y-%m-%dT%H:%M:%SZ' --date=@$SOURCE_DATE_EPOCH) \
-              -extldflags \"$LDFLAGS\"" \
+    -ldflags "${go_ldflags[*]}" \
     .
 }
 
@@ -61,3 +65,5 @@ package() {
 
   install -Dm644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
 }
+
+pkgrel+=.0
Index: b/REUSE.toml
===================================================================
--- a/REUSE.toml
+++ b/REUSE.toml
@@ -2,21 +2,21 @@ version = 1
 
 [[annotations]]
 path = [
-    "PKGBUILD",
-    "README.md",
-    "keys/**",
-    ".SRCINFO",
-    ".nvchecker.toml",
-    "*.install",
-    "*.sysusers",
-    "*.tmpfiles",
-    "*.logrotate",
-    "*.pam",
-    "*.service",
-    "*.socket",
-    "*.timer",
-    "*.desktop",
-    "*.hook",
+  "PKGBUILD",
+  "README.md",
+  "keys/**",
+  ".SRCINFO",
+  ".nvchecker.toml",
+  "*.install",
+  "*.sysusers",
+  "*.tmpfiles",
+  "*.logrotate",
+  "*.pam",
+  "*.service",
+  "*.socket",
+  "*.timer",
+  "*.desktop",
+  "*.hook",
 ]
 SPDX-FileCopyrightText = "Arch Linux contributors"
 SPDX-License-Identifier = "0BSD"
