Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Sun, 21 Sep 2025 20:46:08 -0300
Last-Update: 2026-05-01
Subject: [PATCH] Use latest master, which includes <https://gitlab.com/CalcProgrammer1/OpenRGB/-/merge_requests/2999>
---
 .SRCINFO   |    7 ++++---
 .gitignore |    2 ++
 PKGBUILD   |   14 +++++++-------
 3 files changed, 13 insertions(+), 10 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,11 +1,12 @@
 pkgbase = openrgb
 	pkgdesc = Open source RGB lighting control that doesn't depend on manufacturer software
 	pkgver = 1.0rc2
-	pkgrel = 6
+	pkgrel = 6.3
 	url = https://gitlab.com/CalcProgrammer1/OpenRGB
 	arch = x86_64
 	license = GPL-2.0-or-later
 	makedepends = qt5-tools
+	makedepends = git
 	depends = glibc
 	depends = libgcc
 	depends = libstdc++
@@ -15,11 +16,11 @@ pkgbase = openrgb
 	depends = mbedtls
 	depends = hicolor-icon-theme
 	optdepends = i2c-tools: mainboard & RAM access
-	source = https://gitlab.com/CalcProgrammer1/OpenRGB/-/archive/release_candidate_1.0rc2/OpenRGB-release_candidate_1.0rc2.tar.gz
+	source = git+https://gitlab.com/CalcProgrammer1/OpenRGB.git#commit=92a9d8ad117f100b636f941739058f31730cf84f
 	source = openrgb-modules-load.conf
 	source = openrgb.tmpfiles
 	source = openrgb.service
-	sha256sums = 7ecef99ab1607bd8f47427eef59e060e407dc97a40bb9083c5ade6f30145b762
+	sha256sums = 5ec1970c7ed453bd26dea3ca9e56a1c38252890f1f3e965fc52804e625cbfead
 	sha256sums = b5a53d747422f8b594e3e9615e238457d696732efce94050cdd72182a8645ef2
 	sha256sums = e86604e9604d59cddcc13de60b71c0e652cd1f9b56d89de36b2158509849e51c
 	sha256sums = f53328babef392393ee44086f8897174232e863280a84266630a3cd92facc190
Index: b/.gitignore
===================================================================
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,2 @@
+*.pkg.tar.*
+OpenRGB
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -3,19 +3,19 @@
 # Contributor: Paul Davis <paul@dangersalad.com>
 pkgname=openrgb
 pkgver=1.0rc2
-pkgrel=6
+pkgrel=6.3
 pkgdesc="Open source RGB lighting control that doesn't depend on manufacturer software"
 arch=("x86_64")
 url="https://gitlab.com/CalcProgrammer1/OpenRGB"
 license=('GPL-2.0-or-later')
 depends=('glibc' 'libgcc' 'libstdc++' 'qt5-base' 'libusb' 'hidapi' 'mbedtls' 'hicolor-icon-theme')
-makedepends=('qt5-tools')
+makedepends=('qt5-tools' 'git')
 optdepends=('i2c-tools: mainboard & RAM access')
-source=("https://gitlab.com/CalcProgrammer1/OpenRGB/-/archive/release_candidate_$pkgver/OpenRGB-release_candidate_${pkgver}.tar.gz"
+source=("git+$url.git#commit=92a9d8ad117f100b636f941739058f31730cf84f"
         openrgb-modules-load.conf
         openrgb.tmpfiles
         openrgb.service)
-sha256sums=('7ecef99ab1607bd8f47427eef59e060e407dc97a40bb9083c5ade6f30145b762'
+sha256sums=('5ec1970c7ed453bd26dea3ca9e56a1c38252890f1f3e965fc52804e625cbfead'
             'b5a53d747422f8b594e3e9615e238457d696732efce94050cdd72182a8645ef2'
             'e86604e9604d59cddcc13de60b71c0e652cd1f9b56d89de36b2158509849e51c'
             'f53328babef392393ee44086f8897174232e863280a84266630a3cd92facc190')
@@ -27,18 +27,18 @@ build() {
   # Fix crashes
   export CXXFLAGS=${CXXFLAGS/-Wp,-D_GLIBCXX_ASSERTIONS}
 
-  cd "$srcdir/OpenRGB-release_candidate_$pkgver"
+  cd "$srcdir/OpenRGB"
   sed -i 's|rules.path=/lib|rules.path=/usr/lib|g' OpenRGB.pro
   qmake OpenRGB.pro
   make
 }
 
 package() {
-  cd "$srcdir/OpenRGB-release_candidate_$pkgver"
+  cd "$srcdir/OpenRGB"
   make INSTALL_ROOT="$pkgdir" install
   install -Dm644 "$srcdir"/openrgb-modules-load.conf "$pkgdir"/usr/lib/modules-load.d/openrgb.conf
   install -Dm644 "$srcdir"/openrgb.tmpfiles "$pkgdir"/usr/lib/tmpfiles.d/openrgb.conf
   install -Dm644 "$srcdir"/openrgb.service "$pkgdir"/usr/lib/systemd/system/openrgb.service
   # Remove the upstream service file sicne we prefer ours
-  rm -r "$pkgdir"/etc
+  rm -rf "$pkgdir"/etc
 }
