Author: Tiago de Paula <tiagodepalves@gmail.com>
Date: Sun, 25 Jan 2026 05:08:28 -0300
Last-Update: 2026-04-07
Subject: [PATCH] Update to <https://gitlab.gnome.org/GNOME/sushi/-/tree/30986f21> (2026-02-23)
---
 .SRCINFO   |    4 ++--
 .gitignore |    3 +++
 PKGBUILD   |    4 ++--
 3 files changed, 7 insertions(+), 4 deletions(-)

Index: b/.SRCINFO
===================================================================
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = sushi
 	pkgdesc = A quick previewer for Nautilus
 	pkgver = 50rc.1
-	pkgrel = 1
+	pkgrel = 1.2
 	url = https://gitlab.gnome.org/GNOME/sushi
 	arch = x86_64
 	groups = gnome
@@ -28,7 +28,7 @@ pkgbase = sushi
 	depends = pango
 	optdepends = libreoffice: OpenDocument formats
 	optdepends = webkit2gtk-4.1: Render HTML files
-	source = git+https://gitlab.gnome.org/GNOME/sushi.git?signed#tag=50.rc.1
+	source = git+https://gitlab.gnome.org/GNOME/sushi.git#commit=30986f2168a651bb5157cf5bbd6f826684535adc
 	source = 0001-sushi-media-bin-Fix-llvmpipe-detection.patch
 	validpgpkeys = 550660707A6F40376B9B9F8D504A78811E6160CC
 	b2sums = db457838b473a074bef4ba38fe5ca715255cc8bba92470b56169e6d97e5136761ddc0d71d1ff3a7372ddfbf73829746d9581b5b268bd58e3ae36e07e955a5a3c
Index: b/.gitignore
===================================================================
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,3 @@
+*.pkg.tar.*
+*.log
+sushi/
Index: b/PKGBUILD
===================================================================
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -4,7 +4,7 @@
 
 pkgname=sushi
 pkgver=50rc.1
-pkgrel=1
+pkgrel=1.2
 pkgdesc="A quick previewer for Nautilus"
 url="https://gitlab.gnome.org/GNOME/sushi"
 arch=(x86_64)
@@ -39,7 +39,7 @@ optdepends=(
 )
 groups=(gnome)
 source=(
-  "git+$url.git?signed#tag=${pkgver/[a-z]/.&}"
+  "git+$url.git#commit=30986f2168a651bb5157cf5bbd6f826684535adc"
   0001-sushi-media-bin-Fix-llvmpipe-detection.patch
 )
 b2sums=('db457838b473a074bef4ba38fe5ca715255cc8bba92470b56169e6d97e5136761ddc0d71d1ff3a7372ddfbf73829746d9581b5b268bd58e3ae36e07e955a5a3c'
